library/bignum.c

```diff
@@ -2231,7 +2260,8 @@ int mbedtls_mpi_gen_prime( mbedtls_mpi *X, size_t nbits, int dh_flag,
 
         if( dh_flag == 0 )
         {
-            ret = mbedtls_mpi_is_prime( X, f_rng, p_rng );
+            ret = mbedtls_mpi_is_prime_internal( X, MPI_FIPS_COMPLIANCE, f_rng,
```

Comment:
Why does `dh_flag` enable `MPI_FIPS_COMPLIANCE`? I don't see a connection between “safe prime for Diffie-Hellman” and “FIPS compliance”.

Reply:
As far as I can tell there are no general FIPS restrictions for DH key exchange like there are for RSA digital signatures. And even if there were, it is out of scope of this PR.